File: modelo/rolesusuario.cs
using System.Data;
using System;
    public class rolesusuario : oObjeto
    {
        public int Idroles{ get; set; }
        public int Idusuario { get; set; }
        public string Descripcion{set;get;}
        public rolesusuario(){}
        public rolesusuario(DataRow dr)
    {
        //this.Id = Convert.ToInt32(dr["id"]);
        this.Idroles= Convert.ToInt32(dr["idroles"]);
        this.Idusuario= Convert.ToInt32(dr["idusuario"]);
        this.Descripcion= dr["descripcion"].ToString();
    }

    }
 

File: ClientApp/src/app/clases/rolesusuario.ts
export class rolesusuario{
    id : number; 
    idroles : number;
    idusuario : number;
    descripcion : string;
   
    
    constructor(obj:any){
        //el orden de os parametros afecta la vista de los input en el modal
        this.id = +obj.id;
        this.idroles = +obj.idroles;
        this.idusuario =  +obj.idusuario;
        this.descripcion = obj.descripcion;
       
         // formateo a number el string // parseInt(val

    }
    
}